### DISCUSSION TITLE
New Trigger: MovePunished

### DISCUSSION THREAD

**Discussion by bluish-elvis** [ikemen-engine/Ikemen-GO#2130] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

Are there any plans to add the trigger of player has Punished Opponent ?

### Describe the solution you'd like

The trigger works like MoveCountered, when p1 interrupts p2's attack-state **but It's after shown Clsn1 with Hitdef/Reversaldef, or once Projectile fired in the state**().

> Format:
> MovePunished
> 
> Arguments:
> none
> 
> Return type:
> int

trigger1 = MovePunished = 1

**Comment by potsmugen** [comment, marked as answer]
This is gray enough that it can't/shouldn't be a native trigger. For one the punish mechanic can have different requirements by game (e.g. invincible moves in SFV). Then for it to work properly everyone would have to code attack recovery the same way (flagged as movetype I).

With that said it's already possible to implement such systems with char code.
